index.js

```diff
@@ -41,7 +46,7 @@ module.exports = {
 		var handler;
 		if (arguments.length === 3) {
 			handler = function(req, res) {
-				if (req.method() === method && req.url() === url) {
+				if (req.method() === method && req.url() === _getBaseUrl(url)) {
```

Comment:
It think @emirotin's approach is best, I realised I needed to change my URL matching to use a regexp to extract parts to of some REST API calls.